storage/database/dynamodb.go

```diff
@@ -474,7 +487,7 @@ func (batch *dynamoBatch) Put(key, val []byte) error {
 
 	if len(batch.batchItems) == dynamoBatchSize {
 		batch.wg.Add(1)
-		batch.db.writeCh <- &batchWriteWorkerInput{batch.batchItems, batch.wg}
+		dynamoWriteCh <- &batchWriteWorkerInput{batch.tableName, batch.batchItems, batch.wg}
```

Comment:
Is there any possibility of panic if `Close` is called while processing `Put` operation?

Reply:
I've just tested.
A panic will not occur even though `Close()` is called on non-empty channel. (call `Put()` -> `Close()`) It will close after all of the items are processed.
However, a panic will occur when put is called in a closed channel (call `Close()` -> `Put()`).

Reply:
Yes. To avoid the second case, we need to use switch & case. Do you have any other alternatives?

Reply:
If we use the original code, not all of the items in the channel will be processed. There might be some items we expect to be writen but actually not written. Actually, in the original code, it will wait inifitively in `Write()` because the workers will not process the input and `wg.Done()` is not called.
Is there other ways to handle both cases?

Reply:
I think we don't need to consider the 2nd case. It is misbehavior of users.

Reply:
> I think we don't need to consider the 2nd case. It is misbehavior of users.

I think this is the case which rarely happens. 
However, I don't think it is misbehavior of users. Since the user does not know whether it will cause panic or not. 

However, again, as `Database.Close` is called after `Blockchain.Stop`, there is no possibility that `Database.Put` is called after `Database.Close`.